src/librustc_mir/monomorphize/partitioning.rs

```diff
@@ -455,11 +455,18 @@ fn default_visibility(tcx: TyCtxt<'_>, id: DefId, is_generic: bool) -> Visibilit
 fn merge_codegen_units<'tcx>(
     tcx: TyCtxt<'tcx>,
     initial_partitioning: &mut PreInliningPartitioning<'tcx>,
-    target_cgu_count: usize,
+    mut target_cgu_count: usize,
 ) {
     assert!(target_cgu_count >= 1);
     let codegen_units = &mut initial_partitioning.codegen_units;
 
+    if tcx.is_compiler_builtins(LOCAL_CRATE) {
+        // Compiler builtins require some degree of control over how mono items
+        // are partitioned into compilation units. Provide it by keeping the
+        // original partitioning when compiling the compiler builtins crate.
```

Comment:
Could this comment be updated to basically indicate that for the compiler-builtins crate specifically we want one-symbol-per-object-file like most C implementations do? And comment as well about how that appeases linkers where the goal is that we're trying to avoid duplicate symbol errors when other compiler-rt libraries probably have the same symbols but highly likely the same API too